onboard/nav/obstacleAvoidanceStateMachine.cpp

```diff
@@ -0,0 +1,84 @@
+#include "obstacleAvoidanceStateMachine.hpp"
+
+#include "utilities.hpp"
+#include "stateMachine.hpp"
+#include "simpleAvoidance.hpp"
+#include <cmath>
+#include <iostream>
+
+// Constructs an ObstacleAvoidanceStateMachine object with roverStateMachine
+ObstacleAvoidanceStateMachine::ObstacleAvoidanceStateMachine( StateMachine* stateMachine_ )
+    : roverStateMachine( stateMachine_ ),
+      mJustDetectedObstacle( false ) {}
+
+// Allows outside objects to set the original obstacle angle
+// This will allow the variable to be set before the rover turns
+void ObstacleAvoidanceStateMachine::updateObstacleAngle( double bearing ) 
+{
+    mOriginalObstacleAngle = bearing;
+}
+
+// Runs the avoidance state machine through one iteration. This will be called by StateMachine 
+// when NavState is in an obstacle avoidance state. This will call the corresponding function based 
+// on the current state and return the next NavState
+NavState ObstacleAvoidanceStateMachine::run( Rover* phoebe, const rapidjson::Document& roverConfig )
+{
+    switch ( phoebe->roverStatus().currentState() )
+    {
+        case NavState::TurnAroundObs:
+        case NavState::SearchTurnAroundObs:
+        {
+            return executeTurnAroundObs( phoebe, roverConfig );
+        }
+
+        case NavState::DriveAroundObs:
+        case NavState::SearchDriveAroundObs:
+        {
+            return executeDriveAroundObs( phoebe );
+        }
+
+        default:
+        {
+            return NavState::Unknown;
```

Comment:
put a `cerr` statement here to help us track down this bug if it were to ever happen